src/app/features/dialogs/cancel-transaction-dialog/hooks/use-stx-cancel-transaction.ts

```diff
@@ -0,0 +1,89 @@
+import { useCallback } from 'react';
+
+import { stxToMicroStx } from '@leather-wallet/utils';
+import { TransactionTypes } from '@stacks/connect';
+import * as yup from 'yup';
+
+import { useRefreshAllAccountData } from '@app/common/hooks/account/use-refresh-all-account-data';
+import {
+  type GenerateUnsignedTransactionOptions,
+  generateUnsignedTransaction,
+} from '@app/common/transactions/stacks/generate-unsigned-txs';
+import { getBurnAddress } from '@app/common/utils/get-burn-address';
+import { safelyFormatHexTxid } from '@app/common/utils/safe-handle-txid';
+import { stxFeeValidator } from '@app/common/validation/forms/fee-validators';
+import { useStacksBroadcastTransaction } from '@app/features/stacks-transaction-request/hooks/use-stacks-broadcast-transaction';
+import { useCurrentStxAvailableUnlockedBalance } from '@app/query/stacks/balance/account-balance.hooks';
+import { useCalculateStacksTxFees } from '@app/query/stacks/fees/fees.hooks';
+import { useCurrentStacksAccount } from '@app/store/accounts/blockchain/stacks/stacks-account.hooks';
+import { useCurrentStacksNetworkState } from '@app/store/networks/networks.hooks';
+import { useSubmittedTransactionsActions } from '@app/store/submitted-transactions/submitted-transactions.hooks';
+import { useRawDeserializedTxState, useRawTxIdState } from '@app/store/transactions/raw.hooks';
+
+import { useSelectedTx } from './use-selected-tx';
+
+export function useStxCancelTransaction() {
+  const [rawTxId, setRawTxId] = useRawTxIdState();
+  const tx = useSelectedTx();
+  const [, setTxId] = useRawTxIdState();
```

Comment:
Made a comment on the duplicated PR, but this code can't just be copied from the increase fee dialog. I have refactored it here bc there is code duplication causing bugs. See changes here: https://github.com/leather-wallet/extension/pull/5500

`useRawTxIdState` is an old atom being deprecated.

Reply:
@fbwoolf in that case i will wait for #5500 and remove redundancy